### PR TITLE
Add verify-pr-labels GitHub workflow

### PR DESCRIPTION
**Pull request by BruceHaley**
Fixes #6437 (https://github.com/microsoft/botframework-sdk/issues/6437)

## Description
Add a GitHub workflow PR check to validate PR parity labels.

## Specific Changes
Add workflow file .github/workflows/verify-pr-labels.yml.

Add the folder actions/verify-pr-labels which contains the source files for the workflow action. 

The actions folder is intended to be the parent folder for other GitHub workflow action source folders as needed.

This GitHub workflow can be replicated in the JS repo by copying the files and folders there and merging. No code changes would be needed. 

## Testing
This was tested using the dummy PR here: https://github.com/BruceHaley/botbuilder-dotnet/pull/9
You can test it yourself by changing the labels for the PR.

### CODE REVIEW

File: .github/workflows/verify-pr-labels.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    if: github.event.pull_request.draft == false
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install github actions dependencies
+        uses: bahmutov/npm-install@v1.6.2
```
Comment (gabog) [inline .github/workflows/verify-pr-labels.yml:18]: Do we need to install npm to run this action? Does it work if we remove this? (I think the ubuntu image should have it by default but I might be wrong)

File: actions/verify-pr-labels/index.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    const octokit = new github.getOctokit(gitHubToken);
+
+    const validLabels = [
+      `Automation: No parity`, 
+      `Automation: Parity all`, 
+      `Automation: Parity with dotnet`, 
+      `Automation: Parity with Java`, 
+      `Automation: Parity with JS`, 
+      `Automation: Parity with Python`,
+      `Automation: Parity with Labels`];
```
Comment (gabog) [inline actions/verify-pr-labels/index.js:29]: What is `Parity with Labels`?

File: actions/verify-pr-labels/index.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    if (prValidLabels.length > 0) {
+      core.info(`OK: Pull Request has at least one parity label.`);
+    }
+    else {
+      core.error(`Required is at least one of these labels: ${validLabels.join(`, `)}`);
+      throw `no labels`;
+    }
+
+    // Ensure no other parity labels accompany a `no parity` label.
+    const parityLabelConflict = prValidLabels.find(element => {
```
Comment (gabog) [inline actions/verify-pr-labels/index.js:58]: `Automation: Parity all` should also be single use like `Automation: No parity`

**Review by gabog — CHANGES_REQUESTED** [review]
See comments and let me know if you have any questions.

File: actions/verify-pr-labels/index.js (hunk 2 above)
Comment (BruceHaley) [reply actions/verify-pr-labels/index.js]: Debug value removed.

File: actions/verify-pr-labels/index.js (hunk 3 above)
Comment (BruceHaley) [reply actions/verify-pr-labels/index.js:57]: Done.

File: actions/verify-pr-labels/index.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    core.debug(`Found PR labels: ${prLabels.toString()}`);
+
+    // Get the valid parity labels in this pull request.
+    const prValidLabels = prLabels.filter(value => validLabels.includes(value));
+
+    if (prValidLabels.length > 0) {
+      core.info(`OK: Pull Request has at least one parity label.`);
+    }
+    else {
+      core.error(`Required is at least one of these labels: ${validLabels.join(`, `)}`);
```
[the remark this answers is not in the document]
Comment (BruceHaley) [reply actions/verify-pr-labels/index.js]: Done.

File: .github/workflows/verify-pr-labels.yml (hunk 1 above)
Comment (BruceHaley) [reply .github/workflows/verify-pr-labels.yml:18]: Test result:
Removing 'uses: bahmutov/npm-install@v1.6.2', the workflow does not run.
Removing the whole 'Install github actions dependencies' step, the workflow errors with a dependency not found.

**Review by gabog — APPROVED** [review]

**Pull request merged**